Make errno protocol compatible with Newlib

pfalcon opened the pull request:
This patch makes it possible to use errno in apps built with CONFIG_NEWLIB_LIBC=y.

Issue found when coding BSD Sockets API.

pfalcon committed:
kernel: errno: Use per-thread accessor function compatible with Newlib

Newlib names this function __errno(), so if we want Zephyr to work
with Newlib seamlessly, it's better to just follow Newlib's naming
convention for Zephyr's own minimal libc.

Signed-off-by: Paul Sokolovsky <paul.sokolovsky@linaro.org>

nashif reviewed (changes requested):
Duplication not required, you can build the original test with a different configuration, just a new entry into the testcase.ini and make it use a different configuration enabling newlib.

pfalcon committed:
tests: kernel: errno: Add Newlib test config.

Added because previously, Zephyr used API incompatible with Newlib
for errno handling. Even with Newlib compatibility changes, we
override the function which is defined in Zephyr SDK libc.a, so
makes sense to ensure thsi works as expected.

Signed-off-by: Paul Sokolovsky <paul.sokolovsky@linaro.org>

pfalcon commented:
@nashif : Reworked per the suggestion.
